issue_processing/jira_issue_processing.py

```diff
@@ -225,6 +259,69 @@ def merge_user_with_user_from_csv(user, persons):
     return issues
 
 
+def load_issue_via_api(issues, persons):
+    """For each issue in the list the history is added via the api
+
+        :param issues: list of issues
+        :param persons: list of persons from JIRA (incl. e-mail addresses)
+    """""
+
+    log.info("Load issue information via api...")
+    jira_project = JIRA(project_url)
+
+    for issue in issues:
+
+        min_date = issue['creationDate']
+        api_issue = jira_project.issue(issue['externalId'], expand='changelog')
+        changelog = api_issue.changelog
+        histories = list()
+
+        for change in changelog.histories:
+
+            old_state, new_state, old_resolution, new_resolution = "None", "None", "None", "None"
+
+            for item in change.items:
+
+                if item.field == 'status':
+                    old_state = str(item.fromString)
+                    new_state = str(item.toString)
+
+                elif item.field == 'resolution':
+                    old_resolution = str(item.fromString)
+                    new_resolution = str(item.toString)
+
+            # only history entries that change the state or the resolution are interesting
+            if (old_state != new_state) | (old_resolution != new_resolution):
+
+                history = dict()
+                history['state'] = new_state
+                history['resolution'] = new_resolution
+                user = dict()
+                user["username"] = change.author.name
+                user["name"] = change.author.name
+                user["email"] = ""
+                history["author"] = merge_user_with_user_from_csv(user, persons)
+                date = change.created
+                d = datetime.strptime(date, "%Y-%m-%dT%H:%M:%S.%f+0000")
+                history['date'] = d.strftime('%Y-%m-%d %H:%M:%S')
+                histories.append(history)
+
+                # updates state and resolution for issue and comments in the past
+                if (min_date <= issue['creationDate']) & (issue['creationDate'] < history['date']):
```

Comment:
I am not sure what you want to achieve in this if-block and whether the condition is appropriate for this task. Can you please elaborate?

Reply:
The if-block examines if the issue creation date is between the last checked history entry and the now checked history entry (history entries are checked in time order from oldest to newest). If this is true the state and resolution of the issue get updated, so that you get the state and the resolution the issue had when it was created. I'm not sure if this is necessary for the issue itself, because there might be default values after creating an issue.

Reply:
As discussed in our Thursday meeting, I would like to track this information differently: the global information about the issue as one part and the event information as the other part. The current state of the issue needs to be incorporated in the latter. (see also the comments on Line 498 below)